src/components/dialog/dialog.ts

```diff
@@ -0,0 +1,270 @@
+import { html, LitElement } from 'lit';
+import { property, query, queryAssignedElements } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { themes } from '../../theming';
+import { watch } from '../common/decorators';
+import { Constructor } from '../common/mixins/constructor';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { styles } from './themes/light/dialog.base.css';
+import { styles as bootstrap } from './themes/light/dialog.bootstrap.css';
+import { styles as fluent } from './themes/light/dialog.fluent.css';
+import { styles as indigo } from './themes/light/dialog.indigo.css';
+import { styles as material } from './themes/light/dialog.material.css';
+
+let nextId = 0;
+
+export interface IgcDialogEventMap {
+  igcOpening: CustomEvent<void>;
+  igcOpened: CustomEvent<void>;
+  igcClosing: CustomEvent<void>;
+  igcClosed: CustomEvent<void>;
+}
+
+/**
+ * Represents a Dialog component.
+ *
+ * @element igc-dialog
+ *
+ * @fires igcOpening - Emitted just before the dialog is open.
+ * @fires igcOpened - Emitted after the dialog is open.
+ * @fires igcClosing - Emitter just before the dialog is closed.
+ * @fires igcClosed - Emitted after closing the dialog.
+ *
+ * @slot - Renders content inside the default slot.
+ * @slot title - Renders the title of the dialog header.
+ * @slot footer - Renders the dialog footer.
+ *
+ * @csspart base - The base wrapper of the dialog.
+ * @csspart title - The title container.
+ * @csspart footer - The footer container.
+ * @csspart overlay - The overlay.
+ */
+@themes({ bootstrap, material, fluent, indigo })
+export default class IgcDialogComponent extends EventEmitterMixin<
+  IgcDialogEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-dialog';
+  public static styles = [styles];
+
+  private titleId = `title-${nextId++}`;
+  private _ariaLabel!: string;
+  private _ariaLabelledby!: string;
+  private _ariaDescribedby!: string;
```

Comment:
The current limitations of referencing nodes between shadow/light DOM in ARIA context, make these kind of useless. I suggest you remove the labelledby and describedby props